src/Compat/CompatForm.php

```diff
@@ -3,21 +3,79 @@
 namespace ipl\Web\Compat;
 
 use http\Exception\InvalidArgumentException;
+use ipl\Html\Contract\FormElement;
 use ipl\Html\Contract\FormSubmitElement;
+use ipl\Html\Contract\HtmlElementInterface;
 use ipl\Html\Form;
+use ipl\Html\FormDecorator\DecoratorChain;
 use ipl\Html\FormElement\SubmitButtonElement;
 use ipl\Html\FormElement\SubmitElement;
 use ipl\Html\HtmlDocument;
 use ipl\Html\HtmlString;
 use ipl\I18n\Translation;
 use ipl\Web\FormDecorator\IcingaFormDecorator;
 
+/**
+ * @phpstan-import-type decoratorsFormat from DecoratorChain
+ */
 class CompatForm extends Form
 {
     use Translation;
 
     protected $defaultAttributes = ['class' => 'icinga-form icinga-controls'];
 
+    protected array $elementDecoratorLoaderPaths = [
+        ['ipl\\Web\\Compat\\FormDecorator', 'Decorator']
+    ];
+
+    public function __construct()
+    {
+        $this->setDefaultElementDecorators($this->getDefaultDecorationSpecs());
+    }
+
+    /**
+     * Get the default decoration specifications
+     *
+     * @return decoratorsFormat
+     */
+    protected function getDefaultDecorationSpecs(): array
+    {
+        return [
+            'Label',
+            [
+                'name' => 'HtmlTag',
+                'options' => [
+                    'tag' => 'div',
+                    'class' => 'control-label-group',
+                    'condition' => function (FormElement $element): bool {
+                        return ! $element instanceof FormSubmitElement
+                            && (! $element instanceof HtmlElementInterface || $element->getTag() !== 'fieldset');
+                    }
+                ]
+            ],
```

Comment:
Do this with a custom label decorator instead. If there's no label, it must default to use `&nbsp;`